src/webots/vrml/WbProtoManager.hpp

```diff
@@ -135,20 +141,24 @@ class WbProtoManager : public QObject {
   // searches for proto model according to:
   // 1. First in the session list (i.e., PROTO discovered by navigating the PROTO tree stemming from a world file
   // 2. If the first fails, it searches among the known official Webots PROTO (proto-list.xml)
-  WbProtoModel *findModel(const QString &modelName, const QString &worldPath, QStringList baseTypeList = QStringList());
+  WbProtoModel *findModel(const QString &modelName, const QString &worldPath, const QString &parentFilePath,
+                          QStringList baseTypeList = QStringList());
 
-  WbProtoModel *readModel(const QString &fileName, const QString &worldPath, const QString &externUrl = QString(),
+  WbProtoModel *readModel(const QString &url, const QString &worldPath, const QString &prefix = QString(),
                           QStringList baseTypeList = QStringList()) const;
```

Comment:
This is an issue already merged in the master branch but we should avoid passing objects by value
```suggestion
                          const QStringList &baseTypeList = QStringList()) const;
```